src/main/java/bio/terra/app/usermetrics/UserLoggingMetrics.java

```diff
@@ -11,38 +13,36 @@
  * API request will get grouped together even if they are set in different methods.
  */
 @Component
+@RequestScope
```

Comment:
🤔  I think request scope is the right approach for this component. However, this change means that this class can now only be used in a request context. Before, if it was referenced outside of a request context (such as in a flight), the data would be stored in a map but never used. Now it will generate a runtime exception.

There are only three uses of this currently and all are with a request scope, but it's something to keep in mind.